romancal/pipeline/exposure_pipeline.py

```diff
@@ -71,66 +75,95 @@ def process(self, input):
+                input_filename = basename(in_file)
+                log.info(f"Input file name: {input_filename}")
+            else:
+                input_filename = None
+
+            # Open the file
+            input = rdm.open(in_file)
+            log.info(f"Processing a WFI exposure {in_file}")
+
+            self.dq_init.suffix = "dq_init"
+            result = self.dq_init(input)
+            if input_filename:
+                result.meta.filename = input_filename
+            result = self.saturation(result)
+
+            # Test for fully saturated data
             if is_fully_saturated(result):
-                # Set all subsequent steps to skipped
-                for step_str in [
-                    "assign_wcs",
-                    "flat_field",
-                    "photom",
-                    "source_detection",
-                ]:
-                    result.meta.cal_step[step_str] = "SKIPPED"
-
-                # Set suffix for proper output naming
-                self.suffix = "cal"
+                log.info("All pixels are saturated. Returning a zeroed-out image.")
+
+                # Return zeroed-out image file (stopping pipeline)
+                return self.create_fully_saturated_zeroed_image(result)
+
+            result = self.linearity(result)
+            result = self.dark_current(result)
+            result = self.jump(result)
+            result = self.rampfit(result)
+
+            # Test for fully saturated data
+            if "groupdq" in result.keys():
+                if is_fully_saturated(result):
+                    # Set all subsequent steps to skipped
+                    for step_str in [
+                        "assign_wcs",
+                        "flat_field",
+                        "photom",
+                        "source_detection",
+                    ]:
+                        result.meta.cal_step[step_str] = "SKIPPED"
+
+                    # Set suffix for proper output naming
+                    self.suffix = "cal"
+                    results.append(result)
 
                 # Return fully saturated image file (stopping pipeline)
-                return result
-
-        result = self.assign_wcs(result)
-        if result.meta.exposure.type == "WFI_IMAGE":
-            result = self.flatfield(result)
-        else:
-            log.info("Flat Field step is being SKIPPED")
-            result.meta.cal_step.flat_field = "SKIPPED"
+                return results
 
-        result = self.photom(result)
+            result = self.assign_wcs(result)
+            if result.meta.exposure.type == "WFI_IMAGE":
+                result = self.flatfield(result)
+            else:
+                log.info("Flat Field step is being SKIPPED")
+                result.meta.cal_step.flat_field = "SKIPPED"
 
-        if result.meta.exposure.type == "WFI_IMAGE":
+            result = self.photom(result)
             result = self.source_detection(result)
```

Comment:
This might be a merge / rebase issue or something, but I think we want to keep this "only do source detection for images" condition?

Reply:
Yes that must have been wiped out. 
The check is also in the steps so it doesn't matter too much. The logic is here to avoid extra setup.